.github/scripts/test_open_data_assets.py

```diff
@@ -0,0 +1,212 @@
+#!/usr/bin/env python3
+# Run tests on our open data assets to confirm that their row counts match
+# the row counts of the underlying Athena tables when grouped by year.
+#
+# Requires one positional argument, the path to a dbt manifest that can be
+# used to parse exposure data representing open data assets and their
+# dependency graph.
+#
+# Example:
+#
+#   cd dbt
+#   python3 ../.github/scripts/test_open_data_assets.py target/manifest.json
+
+import contextlib
+import datetime
+import io
+import json
+import sys
+import typing
+
+import agate
+import requests
+from dbt.cli.main import dbtRunner
+
+# API URL and params that we can use to grab counts by year for each asset
+ASSET_API_URL = "https://datacatalog.cookcountyil.gov/resource/{asset_id}.json"
+ASSET_API_QUERY_PARAMS = {"$query": "SELECT COUNT(*),year GROUP BY year"}
+DBT = dbtRunner()
+
+# Data for the most recent two years are permitted to be slightly different,
+# according to this buffer value
+BUFFER = 0.02
+
+
+def main() -> None:
+    """Entrypoint for the script."""
+    dbt_manifest_path = sys.argv[1]
+    with open(dbt_manifest_path) as dbt_manifest_file:
+        dbt_manifest_json = json.loads(dbt_manifest_file.read())
+
+    diffs: typing.List[typing.List[typing.Dict]] = []
+
+    for exposure in dbt_manifest_json["exposures"].values():
+        asset_name = exposure["label"]
+        asset_url = exposure["url"]
+        asset_id = asset_url.split("/")[-1]
+
+        # For now, assume just one dependency table, since all of our open
+        # data assets currently have a 1:1 relationship with a view.
+        fq_source_model_name = exposure["depends_on"]["nodes"][0]
+
+        # Dependency tables in the DAG are fully qualified, so we have to
+        # strip the prefix to get the schema and tablename.
+        source_model_name = ".".join(fq_source_model_name.split(".")[2:])
```

Comment:
Taken together, you can see that this test requires we make a lot of assumptions about the formatting of our exposures. Another big one is that we're assuming that every exposure needs to be tested, which may or may not be the case in the future. I think this is probably not a big deal given that we can always use the `meta` configuration for our exposures to set more fine-grained configs if in the future we decide we want more flexibility in this system.

Reply:
Makes sense! I added a new exposure attribute `meta.test_row_count` to enable this test in [4dd4714](https://github.com/ccao-data/data-architecture/pull/156/commits/4dd4714624cbca478483b55adf33c457ed160107).